garden_ai/entrypoints.py

```diff
@@ -122,10 +122,10 @@ class EntrypointMetadata(BaseModel):
     models: List[ModelMetadata] = Field(default_factory=list)
     repositories: List[Repository] = Field(default_factory=list)
     papers: List[Paper] = Field(default_factory=list)
-    _target_garden_doi: Optional[str] = PrivateAttr(
-        None
-    )  # Used internally for publishing.
-    _as_step: Optional[Step] = PrivateAttr(None)  # Used internally for publishing.
+    # The PrivateAttrs below are used internally for publishing.
+    _test_functions: List[str] = PrivateAttr(default_factory=list)
```

Comment:
Yeah good question. And basically I'm thinking along the lines of your second comment. 

I think it shouldn't be public in `EntrypointMetadata` because public attrs there get exposed in docs about it, and basically convey that we want users to fill it in. Because we don't want users manually attaching these functions in the notebook, we don't want that attr to be public.

Whereas it being public in `RegisteredEntrypoint` is good and expected, because we're saying client applications like the UI can use that attribute.